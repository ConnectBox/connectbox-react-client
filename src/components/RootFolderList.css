/* Top-level collapsed table */
table.collapsed {
    background-color: #CCEFEB;
}

table.collapsed > tbody > tr {
    float: left;
}
table.collapsed > tbody td {
    padding: 10px !important;
    line-height: normal !important;
    vertical-align: bottom;
    border-top: 0 !important;
}
