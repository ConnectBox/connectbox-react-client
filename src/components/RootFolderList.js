import './FolderList.css'
import './RootFolderList.css'

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import RootFolderListEntry from './RootFolderListEntry'

class RootFolderList extends Component {
    static propTypes = {
        content: PropTypes.array.isRequired,
        contentRoot: PropTypes.string.isRequired,
        displayFolderName: PropTypes.bool.isRequired,
        iconMetadata: PropTypes.object.isRequired
    }

  render () {
    const { content, contentRoot, displayFolderName, iconMetadata } = this.props
    let menuItems = []
    for (let i = 0; i < content.length; i++) {
      menuItems.push(
        <RootFolderListEntry
            contentRoot={contentRoot}
            key={`root${i}`}
            file={content[i]}
            displayFolderName={displayFolderName}
            iconMetadata={iconMetadata}
        />)
    }

    return (
      <div className="table-responsive">
          <table className="table collapsed">
              <tbody>
                  {menuItems}
              </tbody>
          </table>
      </div>
    )
  }
}

export default RootFolderList
