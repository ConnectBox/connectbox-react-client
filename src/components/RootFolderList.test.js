/* global it, expect, */
import React from 'react'
import RootFolderList from './RootFolderList'
import renderer from 'react-test-renderer'

import { MemoryRouter as Router } from 'react-router-dom'

it('Render RootFolderList with folder name', () => {
  const iconMetadata = {ids: [], names: {}}
  const content = [{type: 'file', ext: 'doc', name: 'Test File'}]
  const component = renderer.create(
    <Router>
      <RootFolderList
        contentRoot='/content'
        content={content}
        displayFolderName
        folderPath='foo'
        iconMetadata={iconMetadata} />
    </Router>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('Render RootFolderList without folder name', () => {
  const iconMetadata = {ids: [], names: {}}
  const content = [{type: 'file', ext: 'doc', name: 'Test File'}]
  const component = renderer.create(
    <Router>
      <RootFolderList
        contentRoot='/content'
        content={content}
        displayFolderName={false}
        folderPath='foo'
        iconMetadata={iconMetadata} />
    </Router>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
