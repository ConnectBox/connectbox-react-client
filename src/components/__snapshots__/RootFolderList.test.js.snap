// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RootFolderList with folder name 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table collapsed"
  >
    <tbody>
      <tr>
        <td
          className="centered"
        >
          <a
            href="#/Test File"
            onClick={[Function]}
          >
            <i
              className="fa "
            />
            <span>
              <br />
              Test File
            </span>
            <span
              className="directory"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Render RootFolderList without folder name 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table collapsed"
  >
    <tbody>
      <tr>
        <td
          className="centered"
        >
          <a
            href="#/Test File"
            onClick={[Function]}
          >
            <i
              className="fa "
            />
            <span
              className="directory row_name"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
